pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/PostBuildScanService.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @Override
+    protected String getName() {
+        return ADDON_SCAN_SERVICE_MAP_KEY;
+    }
+
+    @Override
+    public void trigger() {
+        log.info("Addon Configuration Parameters");
+        Map<String, Map<String, ?>> addonsMap = pigConfiguration.getAddons();
+        postBuildScanConfigMap = addonsMap.get(ADDON_SCAN_SERVICE_MAP_KEY);
+        this.validate();
+        ScanServiceDTO scanServiceDTO = createScanServiceDTO();
+        sendScanRequest(scanServiceDTO);
+
+        try {
+            ScanHelper pssaasService = new ScanHelper(
+                    this.getServiceSecretKey(),
+                    this.getServiceSecretValue(),
+                    this.getServiceUrl());
+            pssaasService.client.triggerScan(scanServiceDTO);
+        } catch (WebApplicationException e) {
+            log.error(e.toString());
+        } catch (URISyntaxException e) {
+            log.error(e.toString());
+        }
+    }
+
+    void validateMapKeys(Map<String, ?> map) throws ConfigMissingException {
+        for (Map.Entry<String, ?> entry : map.entrySet()) {
+            String key = entry.getKey();
+            if (!masterSet.contains(key)) {
+                Validate.fail("'" + key + "' is not a valid build-config.yaml parameter");
+            }
+            Object value = entry.getValue();
+            if (value instanceof List) {
+                List<?> lst = (List<?>) value;
+                if (lst.get(0) instanceof Map) {
+                    int i = 0;
+                    while (i < lst.size()) {
+                        Map<String, ?> subMap = (Map<String, ?>) lst.get(i);
+                        validateMapKeys(subMap);
+                        i++;
+                    }
+                }
+            } else if (value instanceof Map) {
+                Map<String, ?> subMap = (Map<String, ?>) value;
+                validateMapKeys(subMap);
+            }
+        }
+    }
+
+    void checkPresenceOfMandatoryFields(Map<String, ?> map) throws ConfigMissingException {
+        for (String item : mandatoryItems) {
+            List<String> items = map.keySet()
+                    .stream()
+                    .filter(k -> k.trim().contains(item))
+                    .collect(Collectors.toList());
+            if (items.size() != 1) {
+                Validate.fail(" mandatory field '" + item + "' not provided");
+            }
+        }
+    }
+
+    /* Validate our addon config */
+    @Override
+    public void validate() {
+        try {
+            validateMapKeys(postBuildScanConfigMap);
+            checkPresenceOfMandatoryFields(postBuildScanConfigMap);
+            Validate.validateUrl((String) postBuildScanConfigMap.get("serviceUrl"), "PSSaaS");
+        } catch (ConfigMissingException e) {
+            // We can't handle incorrect configuration, terminate the app with a runtime exception
+            throw new RuntimeException(e);
```

Comment:
ConfigMissingException already extends RuntimeException. Why is it being wrapped here?